geometric_kernels/spaces/product.py

```diff
@@ -188,6 +188,7 @@ def __call__(self, X: B.Numeric, **parameters) -> B.Numeric:
             axis=-1
         )  # [N, M, S] --> [N, M]
 
+    @property
```

Comment:
This will break num_eigenfunctions() calls, right? Didn't we have those? Is this done to unify the overall style?